pulser-simulation/pulser_simulation/simulation.py

```diff
@@ -348,7 +348,7 @@ def set_initial_state(
                     "Incompatible shape of initial state."
                     + f"Expected {legal_shape}, got {shape}."
                 )
-            self._initial_state = qutip.Qobj(state, dims=legal_dims)
+            self._initial_state = qutip.Qobj(state, dims=legal_dims).to("CSR")
```

Comment:
As mentioned in a standalone comment, it would be nice to be able to set this.
Also, doesn't this have to be done for the case above as well?

Reply:
About the case above, I am making the tensor product of objects that I know are CSR, so I know the output is CSR :) 

I suggest to go back to how it was before: 
- If no initial state is provided, we store the initial state in CSR (the initial state is the ground state |gg...g>, so very sparse). 
- If the user provides an initial state, then he is going to provide a numpy array in the EmulatorConfig, and the initial state will be stored as Dense. We could then have an optional argument in `run` of `QutipEmulator` and `QutipBackend` like `initial_state_type` that would be by default None (meaning, don't perform any conversion), but could be a str/a qutip.core.data type.

Yet, I am not sure how much this is relevant to care too much about, and how that would fit with the backend API @HGSilveri (notably, how that fits with the user defining the initial state in the new Config class)

Reply:
The new backend API has room for this, but I definitely would leave this to a future PR